eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -1072,30 +1072,30 @@ class Channel(val nodeParams: NodeParams, val wallet: OnChainChannelFunder with
-        //  - *and* an older version of the funding tx confirmed and reached min depth (it won't be re-orged out)
-        //
-        // This means that:
-        //  - the whole current commitment tree has been double-spent and can safely be forgotten
-        //  - from now on, we only need to keep track of the commitment associated to the funding tx that got confirmed
-        //
-        // Force-closing is our only option here, if we are in this state the channel was closing and it is too late
-        // to negotiate a mutual close.
-        log.info("channelId={} was confirmed at blockHeight={} txIndex={} with a previous funding txid={}", d.channelId, w.blockHeight, w.txIndex, w.tx.txid)
-        watchFundingSpent(commitments1.latest.commitment)
-        context.system.eventStream.publish(TransactionConfirmed(d.channelId, remoteNodeId, w.tx))
-        val commitTx = commitments1.latest.fullySignedLocalCommitTx(keyManager).tx
-        val localCommitPublished = Closing.LocalClose.claimCommitTxOutputs(keyManager, commitments1.latest, commitTx, nodeParams.currentBlockHeight, nodeParams.onChainFeeConf, d.finalScriptPubKey)
-        val d1 = DATA_CLOSING(commitments1, d.waitingSince, d.finalScriptPubKey, mutualCloseProposed = Nil, localCommitPublished = Some(localCommitPublished))
-        stay() using d1 storing() calling doPublish(localCommitPublished, commitments1.latest)
+      acceptFundingTxConfirmed(w, d) match {
+        case Right(commitments1) =>
+          if (d.commitments.latest.fundingTxId == w.tx.txid) {
+            // The best funding tx candidate has been confirmed, alternative commitments have been pruned
+            stay() using d.copy(commitments = commitments1) storing()
+          } else {
+            // This is a corner case where:
+            //  - we are using dual funding
+            //  - *and* the funding tx was RBF-ed
+            //  - *and* we went to CLOSING before any funding tx got confirmed (probably due to a local or remote error)
+            //  - *and* an older version of the funding tx confirmed and reached min depth (it won't be re-orged out)
+            //
+            // This means that:
+            //  - the whole current commitment tree has been double-spent and can safely be forgotten
+            //  - from now on, we only need to keep track of the commitment associated to the funding tx that got confirmed
+            //
+            // Force-closing is our only option here, if we are in this state the channel was closing and it is too late
+            // to negotiate a mutual close.
+            log.info("channelId={} was confirmed at blockHeight={} txIndex={} with a previous funding txid={}", d.channelId, w.blockHeight, w.txIndex, w.tx.txid)
+            val commitment = commitments1.latest
+            val d1 = d.copy(commitments = commitments1)
+            spendLocalCurrent(d1)
```

Comment:
Note here how we are simplifying previous code. This isn't really related to the core of the PR.